src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCspProvider.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.using System;
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.Setup;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted
+{
+    public class SQLSetupStrategyCspProvider : SQLSetupStrategy
+    {
+        private const int KeySize = 2048;
+
+        private readonly List<CspParameters> _cspKeyParameters = new List<CspParameters>();
+
+        public SqlColumnEncryptionCspProvider CspProvider { get; }
+
+        public SQLSetupStrategyCspProvider(CspParameters cspParameters)
+            : base(cspParameters.ProviderName + "/" + cspParameters.KeyContainerName)
+        {
+            // Create a new instance of RSACryptoServiceProvider to generate 
+            // a new key pair.  Pass the CspParameters class to persist the  
+            // key in the container.
+            using RSACryptoServiceProvider rsaAlg = new RSACryptoServiceProvider(KeySize, cspParameters);
+            rsaAlg.PersistKeyInCsp = true;
+
+            _cspKeyParameters.Add(cspParameters);
```

Comment:
Am I right that there's only one parameter going in here? Do we need this to be a list?

Reply:
It's not used right now, but I've done this so that it's consistent with CertificateFixtureBase